src/Depends.Core/DependencyAnalyzer.cs

```diff
@@ -154,7 +156,20 @@ public DependencyGraph Analyze(string packageId, string version, string framewor
             }
         }
 
-        public DependencyGraph Analyze(string projectPath, string framework = null)
+        public DependencyGraph AnalyzeSolution(string solution, string framework = null)
+        {
+            var solutionInfo = SolutionFile.Parse(solution);
+            var solutionNode = new SolutionReferenceNode(solution);
+            var builder = new DependencyGraph.Builder(solutionNode);
+            foreach (var project in solutionInfo.ProjectsInOrder.Where(p => p.ProjectType == SolutionProjectType.KnownToBeMSBuildFormat))
```

Comment:
Is it faster if we pass the solution file directly to `AnalyzerManager`? Buildalyzer is able to work with solution files also.

Reply:
It most certainly is :) I even used it "right" [in my other project](https://github.com/diagnostica-stago/SolutionDependencyAnalyzer/blob/master/SolutionDependencyAnalyzer/DependencyAnalyzer.cs#L60) but forgot to do it here